app/helpers/hosts_and_hostgroups_helper.rb

```diff
@@ -48,8 +50,8 @@ def accessible_puppet_proxies(obj)
     list
   end
 
-  def domain_subnets
-    accessible_related_resource(@domain, :subnets)
+  def domain_subnets(type)
+    accessible_related_resource(@domain, :subnets, :query => {:type => type})
```

Comment:
aha, I see. that makes sense, could you please only rename the option from `:query` to `:where` so it is clear this is passed as an additional `where` to the filter? Also, please extract the options from the hash only after the `blank?` short circuit, no point in extracting them if they won't be used.